mleap-serving/src/main/scala/ml/combust/mleap/serving/MleapConfig.scala

```diff
@@ -1,15 +1,14 @@
 package ml.combust.mleap.serving
 
 import com.typesafe.config.Config
+import scala.util.Properties
 
 /**
   * Created by hollinwilkins on 1/31/17.
   */
 case class HttpConfig(config: Config) {
-  val hostname = config.getString("hostname")
-  val port = config.getInt("port")
-  val bindHostname = config.getString("bind-hostname")
-  val bindPort = config.getInt("bind-port")
+  val bindHostname = scala.util.Properties.envOrElse("MLEAP_SERVER_HOSTNAME", config.getString("bind-hostname"))
```

Comment:
See above comments, access the config value directly and rely on typesafe config to interpolate the environment variable.